packages/toolkit/src/components/VersionDropdownSelector.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from "react";
+
+import {
+  Button,
+  cn,
+  Icons,
+  Nullable,
+  Popover,
+  ScrollArea,
+  Tag,
+} from "@instill-ai/design-system";
+
+const truncateStringMiddle = (string: string, maxLength: number) => {
+  if (string.length > maxLength) {
+    const length = Math.floor(maxLength / 2) - 4;
+
+    return `${string.substring(0, length)}...${string.substring(string.length - length)}`;
+  }
+
+  return string;
+};
+
+const VersionButton = ({
+  id,
+  currentVersion,
+  onClick,
+}: {
+  id: string;
+  currentVersion: Nullable<string>;
+  onClick: () => void;
+}) => {
+  return (
+    <Button
+      key={id}
+      className={cn(
+        "w-full !px-2 !py-1.5",
+        currentVersion === id ? "!bg-semantic-bg-secondary" : "",
+      )}
+      variant={"tertiaryColour"}
+      onClick={onClick}
+    >
+      <div className="flex w-full flex-row gap-x-2">
+        <div className="my-auto h-2 w-[9px] rounded-full bg-semantic-secondary-default"></div>
+        <p
+          className={cn(
+            "w-full text-left text-semantic-fg-secondary product-body-text-3-medium text-ellipsis overflow-hidden",
+          )}
+        >
+          {truncateStringMiddle(id, 22)}
+        </p>
+      </div>
+    </Button>
+  );
+};
+
+export type VersionDropdownSelectorProps = {
+  activeVersion: Nullable<string>;
+  versions: string[];
+  onVersionUpdate: (value: string) => void;
+};
+
+export const VersionDropdownSelector = ({
+  activeVersion,
+  versions,
+  onVersionUpdate,
+}: VersionDropdownSelectorProps) => {
+  const [isVersionSelectorOpen, setIsVersionSelectorOpen] =
+    React.useState<boolean>(false);
+
+  return (
+    <div className="relative ml-auto">
+      <div className="absolute bottom-full left-0 ml-4 font-semibold text-sm text-semantic-fg-secondary">
+        Version
+      </div>
+      <Popover.Root
+        onOpenChange={() => setIsVersionSelectorOpen(!isVersionSelectorOpen)}
```

Comment:
@orangecoloured This looks weird. You can write as this instead
```
        onOpenChange={(open) => {
  if (!open) setIsVersionSelectorOpen(open)
}}
```

Reply:
Normally we use this onOpenChange to have the clickOutside event working correctly, which means radix-ui will help use to close the state

So the writing will be list this 

`onOpenChange={(open) => setIsVersionSelectorOpen(open)}`

But sometime, when I only want to borrow the onClickOutside close event but we don't want radix-ui to intervene the open scenario

We can do this 

```
onOpenChange={(open) => {
  if (!open){
     setIsVersionSelectorOpen(open)
  }
}}
```

So it will also activate when the event is to close the popover

Reading from your code, I assume it's the second scenario, am I right?

Reply:
Ok. I misunderstood this prop. Fixed.